File: samples/perl/01table/table.pl
#!/usr/bin/perl
use strict;
use warnings;

use constant LIMIT => 5;

for $a (0 .. LIMIT) {
    if ($a == 0) {
        print("   ");
    } else {
        print(sprintf("%-3d", $a));
    }
    for $b (1 .. LIMIT) {
        if ($a == 0) {
            print(sprintf("%-3d", $b));
        } else {
            print(sprintf("%-3d", $a * $b));
        }
    }
    print("\n");
}

File: samples/rust/01table/table.rs
static LIMIT: i32 = 5;


fn main() {
    for a in 0..LIMIT+1 {
        if a == 0 {
            print!("{:-3}", "");
        } else {
            print!("{:-3}", a);
        }
        for b in 1..LIMIT+1 {
            if a == 0 {
                print!("{:-3}", b);
            } else {
                print!("{:-3}", a*b);
            }
        }
        println!();
    }
}
